lib/chef/provider/package/apt.rb

```diff
@@ -50,86 +56,91 @@ def default_release_options
           "-o APT::Default-Release=#{@new_resource.default_release}" if @new_resource.respond_to?(:default_release) && @new_resource.default_release
         end
 
-        def check_package_state(package)
-          final_installed_version = []
-          final_candidate_version = []
-          final_installed = []
-          final_virtual = []
-
-          [package].flatten.each do |pkg|
-            installed = virtual = false
-            installed_version = candidate_version = nil
-            shell_out!("apt-cache#{expand_options(default_release_options)} policy #{pkg}", {:timeout=>900}).stdout.each_line do |line|
-              case line
-              when /^\s{2}Installed: (.+)$/
-                installed_version = $1
-                if installed_version == '(none)'
-                  Chef::Log.debug("#{@new_resource} current version is nil")
-                  installed_version = nil
-                else
-                  Chef::Log.debug("#{@new_resource} current version is #{installed_version}")
-                  installed = true
-                end
-              when /^\s{2}Candidate: (.+)$/
-                candidate_version = $1
-                if candidate_version == '(none)'
-                  # This may not be an appropriate assumption, but it shouldn't break anything that already worked -- btm
-                  virtual = true
-                  showpkg = shell_out!("apt-cache showpkg #{package}", {:timeout => 900}).stdout
-                  providers = Hash.new
-                  showpkg.rpartition(/Reverse Provides: ?#{$/}/)[2].each_line do |line|
-                    provider, version = line.split
-                    providers[provider] = version
-                  end
-                  # Check if the package providing this virtual package is installed
-                  num_providers = providers.length
-                  raise Chef::Exceptions::Package, "#{@new_resource.package_name} has no candidate in the apt-cache" if num_providers == 0
-                  # apt will only install a virtual package if there is a single providing package
-                  raise Chef::Exceptions::Package, "#{@new_resource.package_name} is a virtual package provided by #{num_providers} packages, you must explicitly select one to install" if num_providers > 1
-                  # Check if the package providing this virtual package is installed
-                  Chef::Log.info("#{@new_resource} is a virtual package, actually acting on package[#{providers.keys.first}]")
-                  installed = check_package_state(providers.keys.first)
-                else
-                  Chef::Log.debug("#{@new_resource} candidate version is #{$1}")
+        def check_package_state(pkg)
+          is_virtual_package = false
```

Comment:
style comment:
not a fan of the shadowing...also, below you used `installed` instead of `is_installed`